### ISSUE TITLE
Upgrading to ES6

### ISSUE THREAD

**Issue by Akxe** [Irrelon/node-irrelon-router#12]
I would be also very good to use new ES6 syntax, it could upgrade performance in many cases. Improve number of needed modules, and would bring new features.

`() => {}` functions to stop from binding this to self
`Promise` to remove dependency on waterfall
`Map` has forEach method, could give you easy aliasing for hostnames
`const, let` to specify scopes and usage
and probably many more

**Comment by Irrelon** [comment]
This is true, having it written with ES6 would be nice, but I challenge the part about performance... the gains (if any) would probably be so small as to make the rewrite pointless unless we are missing a feature in ES5 that we desperately need and is only available in ES6.

I'm not against ES6 per-se, just that if it's not broken, don't fix it comes to mind :)

**Comment by Akxe** [comment]
Loading many libraries can be intense for small server machines, I agree that for normal computer it doesn't really change that much, but for raspberry pi server, it could change a lot

**Comment by Irrelon** [comment]
Certbot is letsencrypt for cli.
